src/main/java/com/google/graphgeckos/dashboard/api/components/BuildInfo.java

```diff
@@ -0,0 +1,69 @@
+package com.google.graphgeckos.dashboard.api.components;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Represents the response json with the aggregated buildbot data.
+ */
+public class BuildInfo {
+
+  // git commit hash
+  private final String commitHash;
+  // time of git commit push
+  private final Date timestamp;
+  // link to to the repository
+  private final String repository;
+  // name of the branch
+  private final String branch;
+
+  // builders
+  private final List<Builder> builders;
+
+  public BuildInfo(String commitHash, Date timestamp, String repository,
```

Comment:
make BuildInfo constructor only package accessible. No other component besides the storage should be able to create such an object. This applies to all other subclasses' constructors